### PR TITLE
Implemented ExponentialSearcher (Algorithms.Search) and tests

### PR DESCRIPTION
**Pull request by gabcoroba**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Search/ExponentialSearcher.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if ((endLocation - startLocation) <= 0)
+            {
+                return -1;
+            }
+
+            int bound = 1;
+
+            while (bound < (endLocation - startLocation))
+            {
+                if (sortedData[bound] <  item)
```
Comment (siriak) [inline Algorithms/Search/ExponentialSearcher.cs:36]: Please add the inverse of this condition to the loop predicate to eliminate `break`

File: Algorithms/Search/ExponentialSearcher.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return searcher.FindIndex(sortedData, item, (bound / 2), min(bound + 1, sortedData.Length));
+        }
+        }
```
Comment (siriak) [inline Algorithms/Search/ExponentialSearcher.cs:48]: Please fix indentation

File: Algorithms.Tests/Search/ExponentialSearcherTests.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            // Arrange
+            var searcher = new ExponentialSearcher<int>();
+            var arrayToSearch = new int[0];
+
+            // Act
+            var actualIndex = searcher.FindIndex(arrayToSearch, itemToSearch);
+
+            // Assert
+            Assert.AreEqual(-1, actualIndex);
```
Comment (siriak) [inline Algorithms.Tests/Search/ExponentialSearcherTests.cs:59]: Please use FluentAssertions here and above

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, but the build didn't start for some reason. Anyway, I can start it manually once the feedback below is addressed, thanks!

File: Algorithms.Tests/Search/ExponentialSearcherTests.cs (hunk 3 above)
Comment (gabcoroba) [reply Algorithms.Tests/Search/ExponentialSearcherTests.cs]: I added FluentAssertions and addressed the other feedback, thanks!

File: Algorithms.Tests/Search/ExponentialSearcherTests.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Algorithms.Tests.Search
+{
+    public static class ExponentialSearcherTests
+    {
+        [Test]
+        public static void FindIndex_ItemPresent_IndexCorrect([Random(1, 1000, 100)] int n)
+        {
+            // Arrange
+            var subject = new ExponentialSearcher();
```
Comment (siriak) [inline Algorithms.Tests/Search/ExponentialSearcherTests.cs:17]:
```suggestion
            var searcher = new ExponentialSearcher();
```

File: Algorithms.Tests/Search/ExponentialSearcherTests.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            var random = Randomizer.CreateRandomizer();
+            var arrayToSearch = Enumerable.Range(0, n).Select(_ => random.Next(0, 1000)).OrderBy(x => x).ToArray();
+            var selectedIndex = random.Next(0, n);
+
+            // Act
+            var actualIndex = searcher.FindIndex(arrayToSearch, arrayToSearch[selectedIndex]);
+
+            // Assert
+            sortedArray[actualIndex].Should().Be(sortedArray[selectedIndex]);
+        }
```
Comment (siriak) [inline Algorithms.Tests/Search/ExponentialSearcherTests.cs:27]:
```suggestion
        }

```

File: README.md [hunk 6]
```diff
@@ -59,6 +59,7 @@ This repository contains algorithms and data structures implemented in C# for ed
     * [A-Star](./Algorithms/Search/AStar/)
     * [Binary Search](./Algorithms/Search/BinarySearcher.cs)
     * [Recursive Binary Search](./Algorithms/Search/RecursiveBinarySearcher.cs)
+    * [Exponential Search](./Algorithms/Search/ExponentialSearch.cs)
```
Comment (siriak) [inline README.md:62]:
```suggestion
    * [Exponential Search](./Algorithms/Search/ExponentialSearcher.cs)
```

File: Algorithms/Search/ExponentialSearcher.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+
```
Comment (siriak) [inline Algorithms/Search/ExponentialSearcher.cs:4]:
```suggestion
using System.Collections.Generic;

```

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix build errors, thanks!

**Review by siriak — COMMENTED** [review]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(22,13): error CS0305: Using the generic type 'RecursiveBinarySearcher<T>' requires 1 type arguments [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(22,52): error CS0305: Using the generic type 'RecursiveBinarySearcher<T>' requires 1 type arguments [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(33,63): error CS0019: Operator '<' cannot be applied to operands of type 'T' and 'T' [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(37,70): error CS0103: The name 'min' does not exist in the current context [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(7,51): error SA1629: Documentation text should end with a period [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(36,14): error SA1513: Closing brace should be followed by blank line [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(40,2): error SA1518: File is required to end with a single newline character [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(33,82): error SA1025: Code should not contain multiple whitespace characters in a row. [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(15,40): error SA1629: Documentation text should end with a period [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(37,57): error SA1119: Statement should not use unnecessary parenthesis [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
  DataStructures.Tests -> /home/travis/build/TheAlgorithms/C-Sharp/DataStructures.Tests/bin/Debug/net5.0/DataStructures.Tests.dll
Build FAILED.
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(22,13): error CS0305: Using the generic type 'RecursiveBinarySearcher<T>' requires 1 type arguments [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(22,52): error CS0305: Using the generic type 'RecursiveBinarySearcher<T>' requires 1 type arguments [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(33,63): error CS0019: Operator '<' cannot be applied to operands of type 'T' and 'T' [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(37,70): error CS0103: The name 'min' does not exist in the current context [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(7,51): error SA1629: Documentation text should end with a period [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(36,14): error SA1513: Closing brace should be followed by blank line [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(40,2): error SA1518: File is required to end with a single newline character [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(33,82): error SA1025: Code should not contain multiple whitespace characters in a row. [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(15,40): error SA1629: Documentation text should end with a period [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]
/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Search/ExponentialSearcher.cs(37,57): error SA1119: Statement should not use unnecessary parenthesis [/home/travis/build/TheAlgorithms/C-Sharp/Algorithms/Algorithms.csproj]

**Comment by gabcoroba** [comment]
Still on it

**Pull request closed without merging**